reminders/models.py

```diff
@@ -30,22 +38,34 @@ class Reminder(models.Model):
         default=ReminderType.WEBSITE,
     )
 
-    def validate_unique_reminder(self, participant, date_time):
-        if Reminder.objects.filter(
-                participant_id__user_id=participant.user_id,
-                participant_id__event_id=participant.event_id,
-                date_time=date_time
-        ):
-            raise ValidationError('reminder already exists')
+    objects = ReminderQuerySet.as_manager()
+
+    class Meta:
+        constraints = [
+            models.UniqueConstraint(fields=['participant_id', 'date_time'], name='unique reminder'),
+            models.CheckConstraint(check=Q(date_time__gte=Now()), name="date_time__gte_currnet_time")
+        ]
 
     def __str__(self):
         return f"{self.participant_id} - {self.date_time}"
 
-    def clean(self):
-        validate_date(self.date_time)
-        self.validate_unique_reminder(self.participant_id, self.date_time)
-        return super().clean()
-
     def save(self, *args, **kwargs):
-        self.clean()
```

Comment:
Because i have used a different kind of verification.
I have found that the Meta class can define verification inside the sql level,
Which means we don't need to use outside functions as verification aids, we can use only that class for checking constraints.

Reply:
Cool!
TBH for me all this 'try...catch' block little bit reduces the readability,
But I can live with it :)